### PR TITLE
Add dialyzer to build

### PR DESCRIPTION
**Pull request by aaronrenner**
This adds dialyzer to the build as well as specs for `Honeydew.queue_spec/2` and `Honeydew.worker_spec/3`.

### CODE REVIEW

File: lib/honeydew.ex [hunk 1]
```diff
@@ -151,6 +162,7 @@ defmodule Honeydew do
   - `Honeydew.queue_spec("my_awesome_queue")`
   - `Honeydew.queue_spec("my_awesome_queue", queue: {MyQueueModule, [ip: "localhost"]}, dispatcher: {Honeydew.Dispatcher.MRU, []})`
   """
+  @spec queue_spec(String.t, [queue_spec_opt]) :: Supervisor.Spec.spec
```
Comment (koudelka) [inline lib/honeydew.ex]: `queue_name` instead of `String.t`, methinks

File: lib/honeydew.ex [hunk 2]
```diff
@@ -2,6 +2,10 @@ defmodule Honeydew do
   alias Honeydew.Job
   require Logger
 
+  @type queue_name :: String.t
```
[the remark this answers is not in the document]
Comment (aaronrenner) [reply lib/honeydew.ex]: Does `{:via, module, term}` also work? I wonder if the type would be `String.t | GenServer.name`.

[the remark this answers is not in the document]
Comment (aaronrenner) [reply lib/honeydew.ex]: Disregard my previous comment... I don't think queue names have anything to do with process lookup.

[the remark this answers is not in the document]
Comment (koudelka) [reply lib/honeydew.ex]: nah, the queue's name isn't the name used in the process registry. it's added to `:pg2` with the name specified by https://github.com/koudelka/honeydew/blob/master/lib/honeydew.ex#L319

technically it's anything that implements `String.Chars`, but it really should just be the above typespec

File: lib/honeydew.ex [hunk 3]
```diff
@@ -196,6 +208,12 @@ defmodule Honeydew do
       supervisor_opts)
   end
 
+  @type worker_spec_opt ::
+    {:num, non_neg_integer} |
+    {:init_retry, non_neg_integer} |
+    {:supervisor_opts, supervisor_opts} |
+    {:nodes, [atom]}
```
Comment (koudelka) [inline lib/honeydew.ex]: you could do `[node]`, just to be super anal. :)

File: lib/honeydew.ex [hunk 4]
```diff
@@ -2,6 +2,11 @@ defmodule Honeydew do
   alias Honeydew.Job
   require Logger
 
+  @type mod_or_mod_args :: module | {module, list}
+  @type node :: atom
```
Comment (koudelka) [inline lib/honeydew.ex]: `node`'s a builtin

File: lib/honeydew.ex [hunk 5]
```diff
@@ -2,6 +2,11 @@ defmodule Honeydew do
...
+  @type queue_name :: String.t | atom | {:global, String.t | atom}
+  @type supervisor_opts :: Keyword.t
```
Comment (koudelka) [inline lib/honeydew.ex]: not `keyword`?

File: lib/honeydew.ex (hunk 4 above)
Comment (aaronrenner) [reply lib/honeydew.ex]: `node_name`?

File: lib/honeydew.ex (hunk 5 above)
Comment (aaronrenner) [reply lib/honeydew.ex]: Sure. I can change it to `keyword`. It seems weird you can do it either way.

Comment (aaronrenner) [reply lib/honeydew.ex]: Just checked through the docs and `Keyword.t` and `keyword` are identical.

Comment (koudelka) [reply lib/honeydew.ex]: i leave the decision to your aesthetics, then :)

File: lib/honeydew.ex (hunk 4 above)
Comment (koudelka) [reply lib/honeydew.ex]: hm, i just see `node()` on https://hexdocs.pm/elixir/typespecs.html

Comment (aaronrenner) [reply lib/honeydew.ex]: Right... 🤦‍♂️

**Comment by koudelka** [comment]
🙇

**Pull request merged**